src/pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -293,8 +293,8 @@ def restore_precision_plugin_state(self) -> None:
             prec_plugin.load_state_dict(self._loaded_checkpoint[prec_plugin.__class__.__qualname__])
 
         # old checkpoints compatibility
-        if "amp_scaling_state" in self._loaded_checkpoint and isinstance(prec_plugin, ApexMixedPrecisionPlugin):
-            prec_plugin.load_state_dict(self._loaded_checkpoint["amp_scaling_state"])
+        if "amp_scaling_state" in self._loaded_checkpoint:
+            rank_zero_warn("This checkpoint contains apex AMP data, but apex support has been removed.")
```

Comment:
Should we remove this as well?

Reply:
Fine with me either way. I thought having a warning could be useful

Reply:
Alternatively we could also have a migration that removes the key and warns that this state gets dropped.

Reply:
Leaving this for a follow-up